src/class/object_users.php

```diff
@@ -480,9 +482,21 @@ function prepareData ($zData = '', $sView = 'list')
             // Submissions...
             if (lovd_isAuthorized('user', $zData['id']) === false) {
                 // Not authorized to view hidden data for this user; so we're not manager and we're not viewing ourselves. Nevermind then.
-                unset($this->aColumnsViewEntry['ownes_']);
+                unset($this->aColumnsViewEntry['ownes_'], $this->aColumnsViewEntry['auth_token_'], $this->aColumnsViewEntry['auth_token_expires_']);
             } else {
-                // Either we're viewing ourselves, or we're manager or up. Like this is easy, because now we don't need to check for the data status of the data.
+                // Either we're viewing ourselves, or we're manager or up.
+
+                // Auth token links. We don't show the token by default.
+                $zData['auth_token_'] = '(<A href="#" onclick="$.get(\'ajax/auth_token.php/' . $zData['id'] . '?view\'); return false;">Show / More information</A>)';
+                if ($zData['auth_token_expires']) {
+                    $tDiff = strtotime($zData['auth_token_expires']) - time();
+                    $sDiff = lovd_convertSecondsToTime(abs($tDiff));
+                    $zData['auth_token_expires_'] = '<SPAN title="' . $zData['auth_token_expires'] . '">' . ($tDiff > 0? 'In ' . $sDiff : 'Expired ' . $sDiff . ' ago') . '</SPAN>';
+                } else {
+                    $zData['auth_token_expires_'] = (!$zData['auth_token']? '' : '- (Never)');
```

Comment:
> Perhaps it's clearer if the expiration is displayed as "Never" instead of "- (Never)".

I was doubting about that, but chose the latter. I don't think I have much of a preference, so if you prefer "Never" we can change it :wink: 


> Another idea is to show everything related to the token in one VE row, like:

We shouldn't show the token in the VE directly. Even though it's an MD5, it's still sensitive information and should, in my opinion, only be shown when really requested.

But yes, we could merge the two, if you prefer that.